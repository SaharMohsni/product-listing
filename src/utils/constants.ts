export const ASC_PRICE = 0;
export const DESC_PRICE = 1;
export const NEW_ADDED = 2;
export const OLD_ADDED = 3;

export const LIMIT_PRODUCTS_BY_Page = 16;

export const sortDataMessages = {
	[ASC_PRICE]: 'Price low to high',
	[DESC_PRICE]: 'Price high to low',
	[NEW_ADDED]: 'New to old',
	[OLD_ADDED]: 'Old to new'
};

export const MIN_PRODUCT_QUANTITY_IN_BASKET = 1;
export const FAKE_DATA = Array.from({ length: 16 }).map(() => {}).map((el, i) => ({
	tags: [ '' ],
	price: i,
	name: 'string',
	description: 'string',
	slug: 'string',
	added: i,
	manufacturer: i,
	itemType: 'string'
}));
export const DELAY_TIME = 3000;
