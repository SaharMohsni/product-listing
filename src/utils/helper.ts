import { ISearchProductsPayload } from '../features/types/products.types';
import { isEmpty } from 'lodash';
export const getURLCleanPath = (pathname: string) => pathname.split('/')[1];

export const serialize = function(obj: any) {
	var str = [];
	for (var p in obj)
		if (obj.hasOwnProperty(p)) {
			str.push(encodeURIComponent(p) + '=' + encodeURIComponent(obj[p]));
		}
	return str.join('&');
};

export const handleNavigationQuery = (params: ISearchProductsPayload) => {
	let { page, sortVariable, sortType } = params;
	let query = params;
	if (!isEmpty(params) && (isEmpty(sortVariable) || isEmpty(sortType))) {
		query = {
			page: page
		};
	} else {
		query = {
			...params
		};
	}

	return `${serialize(query)}`;
};
