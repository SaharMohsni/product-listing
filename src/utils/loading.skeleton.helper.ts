export const fullRowItemSkeleton = (loading:boolean, active = true) => ({
    active,
    title: { rows: 1, width: '100%' },
    paragraph: false,
    loading,
  })
  export const labelSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 1, width: '20%'  },
    loading,
  })
  export const tagSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 1 },
    loading,
  })
  export const imageSkeleton = (loading:boolean, active = true ) => ({
    loading,
  active,
  paragraph: false,
  title:false
  })

  export const descriptionSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: { rows: 3, width: '100%' },
    loading,
  })
  export const shapeSquareBoxSkeleton = (loading:boolean, active = true) => ({
    active,
    title: false,
    paragraph: false,
    loading,
  })
