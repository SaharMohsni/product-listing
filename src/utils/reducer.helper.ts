import { ISearchProductsPayload } from '../features/types/products.types';

export const handleSearchParams = (localSearchParams: ISearchProductsPayload, params: ISearchProductsPayload) => {
	let { page, limit, sortVariable, sortType } = localSearchParams;
	return {
		...localSearchParams,
		page: params.page ? params.page : page,
		limit: params.limit ? params.limit : limit,
		sortVariable: params.sortVariable ? params.sortVariable : sortVariable,
		sortType: params.sortType ? params.sortType : sortType
	};
};
