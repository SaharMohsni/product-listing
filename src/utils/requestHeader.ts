 
export const requestHeader = (options: object| null ): object => ({
  headers: { 
    // 'Access-Control-Allow-Origin': '*', 
    ...options,
  },
})

 