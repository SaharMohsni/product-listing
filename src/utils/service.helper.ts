import { isEmpty } from 'lodash';
export const addSortToUrl = (variable: string | undefined, sortType: string | undefined) => {
	if (!isEmpty(variable) && !isEmpty(sortType)) {
		return `&_sort=${variable}&_order=${sortType}`;
	} else return '';
};

export const addFilterToUrl = (variable: any | undefined) => {
	let str = [];
	if (variable) {
		for (var el in variable.data) {
			str.push(variable.key + '=' + variable.data[el]);
		}
	}
	return str.join('&');
};

export const addFilterByArrayListToUrl = (variable: any | undefined) => {
	let str = [];
	if (variable) {
		for (var el in variable.data) {
			str.push(variable.key + '_like=' + variable.data[el]);
		}
	}
	return str.join('&');
};

export const addFilterByStringToUrl = (variable: any | undefined) => {
	if (!isEmpty(variable)) {
		return `itemType=${variable}`;
	} else return '';
};
