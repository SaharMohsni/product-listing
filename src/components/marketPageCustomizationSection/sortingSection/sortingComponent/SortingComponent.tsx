import React, { useState } from 'react';
import { Radio, Space, Skeleton } from 'antd';

import './sorting-component.css';
import * as skeleton from '../../../../utils/loading.skeleton.helper';

const SortingComponent = () => {
	const [ sortingValue, setSortingValue ] = useState(1);
	const handleChange = (e: any) => {
		console.log('change');
		setSortingValue(e.target.value);
	};

	return (
		<div className="sorting-component ">
			<Skeleton {...skeleton.labelSkeleton(false)}>
				<div className="section-block-header">Sorting</div>
			</Skeleton>
			<Skeleton avatar={{ shape: 'square' }} {...skeleton.shapeSquareBoxSkeleton(false)}>
				<div className="sorting-component__data section-block box">
					<Radio.Group onChange={(e) => handleChange(e)} value={sortingValue}>
						<Space direction="vertical">
							<Radio value={1}>Price low to high</Radio>
							<Radio value={2}>Price high to low</Radio>
							<Radio value={3}>New to old</Radio>
							<Radio value={4}>Old to new</Radio>
						</Space>
					</Radio.Group>
				</div>
			</Skeleton>
		</div>
	);
};

export default SortingComponent;
