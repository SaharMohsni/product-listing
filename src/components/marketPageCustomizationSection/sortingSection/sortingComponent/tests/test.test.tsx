import React from 'react';
import { render } from '../../../../../utils/testing';
import SortingComponent from '../SortingComponent';

describe('SortingComponent', () => {
	const { container } = render(<SortingComponent />);
	test('should render the basic template', async () => {
		expect(container).toMatchSnapshot();
	});
});
