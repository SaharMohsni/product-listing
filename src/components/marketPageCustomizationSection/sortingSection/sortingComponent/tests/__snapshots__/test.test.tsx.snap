// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingComponent should render the basic template 1`] = `
<div>
  <div
    class="sorting-component "
  >
    <div
      class="section-block-header"
    >
      Sorting
    </div>
    <div
      class="sorting-component__data section-block box global-scroll"
    >
      <div>
        <div
          class="ant-radio-group ant-radio-group-outline"
        >
          <div
            class="ant-space ant-space-vertical"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="0"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  Price low to high
                </span>
              </label>
            </div>
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="1"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  Price high to low
                </span>
              </label>
            </div>
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="2"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  New to old
                </span>
              </label>
            </div>
            <div
              class="ant-space-item"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="3"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  Old to new
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
